### ISSUE TITLE
Universe polymorphism

### ISSUE THREAD

**Issue by robrix** [antitypical/Manifold#42]

**Comment by robrix** [comment]
[Idris’ approach](http://idris.readthedocs.org/en/latest/faq/faq.html#does-idris-have-universe-polymorphism-what-is-the-type-of-type) is to instead use a cumulative hierarchy of types `Type n` where `x` : `Type n` implies `x` : `Type m` iff `n <= m`.

**Comment by robrix** [comment]
Fixed by #62.

**Comment by robrix** [comment]
This isn’t actually fixed by #62. We have a hierarchy of types but they aren’t polymorphic; so while you can write `Type₁` you cannot write `n → Type n`.

Agda’s approach is to use a type `Level` by which `Set` is parameterized, approximately like this:

``` agda
data Level : Set where
    lzero : Level
    lsucc : (n : Level) → Level
    _⨆_ : (n m : Level) → Level -- least upper bound of two levels

data Set : (l : Level) → Set (lsucc l) where
    Set₀ : Set (lsucc lzero)
```

**Comment by robrix** [comment]
Worth noting that Idris’ approach is not implementing universe polymorphism, it’s an alternative to it.
